include/effects/matrix/PatternStockTicker.h

```diff
@@ -0,0 +1,767 @@
+     */
+    size_t DesiredFramesPerSecond() const override
+    {
+        return 10;
+    }
+
+    /**
+     * @brief Does this effect requre double buffering support?
+     * 
+     * @return bool - true 
+     */
+    bool RequiresDoubleBuffering() const override
+    {
+        return true;
+    }
+
+    /**
+     * @brief Retrieve the 'static' information about the supplied
+     * stock symbol
+     * 
+     * @param ticker Reference to a StockTicker object
+     * @return bool - true if the symbol is found by the API
+     */
+    bool updateTickerCode(StockTicker &ticker)
+    {
+        HTTPClient http;
+        String url;
+        bool dataFound = false;
+
+        url = "https://finnhub.io/api/v1/stock/profile2"
+              "?symbol=" + urlEncode(ticker._strSymbol) + "&token=" + urlEncode(g_ptrSystem->DeviceConfig().GetStockTickerAPIKey());
+
+        http.begin(url);
+        int httpResponseCode = http.GET();
+
+        if (httpResponseCode > 0)
+        {
+            /* Sample returned data
+            {   "country":"US",
+                "currency":"USD",
+                "estimateCurrency":"USD",
+                "exchange":"NASDAQ NMS - GLOBAL MARKET",
+                "finnhubIndustry":"Retail",
+                "ipo":"1997-05-15",
+                "logo":"https://static2.finnhub.io/file/publicdatany/finnhubimage/stock_logo/AMZN.svg",
+                "marketCapitalization":1221092.7955074026,
+                "name":"Amazon.com Inc",
+                "phone":"12062661000.0",
+                "shareOutstanding":10260.4,
+                "ticker":"AMZN",
+                "weburl":"https://www.amazon.com/"}
+            */
+
+            AllocatedJsonDocument doc(1024);
+            String headerData = http.getString();
+            debugI("Stock Heder: %s", headerData.c_str());
+            if (headerData.equals("{}")) 
+            {
+                ticker._strCompanyName    = "Bad Symbol";
+                ticker._strExchangeName   = "";
+                ticker._strCurrency       = "";
+                ticker._isValid           = false;
+                ticker._marketCap         = 0.0;
+                ticker._sharesOutstanding = 0.0;
+
+                debugW("Bad ticker symbol: '%s'", ticker._strSymbol.c_str());
+            }
+            else
+            {
+                deserializeJson(doc, headerData);
+                JsonObject companyData =  doc.as<JsonObject>();
+                dataFound = true;
+
+                ticker._strSymbol         = companyData["ticker"].as<String>();
+                ticker._strCompanyName    = companyData["name"].as<String>();
+                ticker._strExchangeName   = companyData["exchange"].as<String>();
+                ticker._strCurrency       = companyData["currency"].as<String>();
+                ticker._marketCap         = companyData["marketCapitalization"].as<float>();
+                ticker._sharesOutstanding = companyData["shareOutstanding"].as<float>();

```

Comment:
I am not as familure with Arduino.Json and did it to ensure that it was what I wanted and so that all the references were consistent.  As this was the first time I had used the library I thought better safe than sorry.  Less runtime work for the library.

I will drop the calls.